ipykernel/ipkernel.py

```diff
@@ -274,21 +274,22 @@ def do_complete(self, code, cursor_pos):
 
     def _experimental_do_complete(self, code, cursor_pos):
         """
-        Experimental completions from IPython, using Jedi. 
+        Experimental completions from IPython, using Jedi.
         """
         if cursor_pos is None:
             cursor_pos = len(code)
         with provisionalcompleter():
             raw_completions = self.shell.Completer.completions(code, cursor_pos)
             completions = list(rectify_completions(code, raw_completions))
-            
+
             comps = []
             for comp in completions:
                 comps.append(dict(
                             start=comp.start,
                             end=comp.end,
                             text=comp.text,
                             type=comp.type,
+                            signature=getattr(comp, 'signature', '')
```

Comment:
+1, one thing I was thinking about is to give that a name more generic name (`right_extra`?) to see how that would extend to other languages. I can't really think of any language where signature is on the left, or maybe language that have `operators` would want to send extra text to display on the left and right ? and call that `extra_left` and `extra_right` ?